src/main/webapp/app/course/plagiarism-cases/instructor-view/detail-view/plagiarism-case-instructor-detail-view.component.ts

```diff
@@ -46,8 +46,13 @@ export class PlagiarismCaseInstructorDetailViewComponent implements OnInit, OnDe
                 this.createEmptyPost();
             },
         });
-        this.postsSubscription = this.metisService.posts.pipe().subscribe((posts: Post[]) => {
-            this.posts = posts;
+        this.postsSubscription = this.metisService.posts.subscribe((posts: Post[]) => {
+            const filteredPosts = posts?.filter((post) => post.plagiarismCase?.id === this.plagiarismCaseId);
+
+            // Handle post deletion case by checking if unfiltered posts are empty.
+            if (filteredPosts.length > 0 || posts.length === 0) {
+                this.posts = filteredPosts;
+            }
```

Comment:
```suggestion
            if (filteredPosts.length > 0) {
                this.posts = filteredPosts;
            }
```
If the length of posts and filteredPosts is both 0, I don't see a reason, why we should override the variable

Reply:
Why does it matter if `posts.length > 0` is the case, when you check `filteredPosts.length >= 0`?
You would simply override the posts with the filteredPosts in that case

Reply:
Probably the version below would be more explicit, I tried to be concise and made use of the fact that if `posts.length === 0` then `filteredPosts.length` also be 0 and it won't make a difference which array is assigned. But I can change it with the version below if that seems better.
```
 if (posts.length === 0) {
    this.posts = posts;
    return;
}
 if (filteredPosts.length > 0) {
    this.posts = filteredPosts;
}
```

Reply:
But if `posts.length === 0` implies `filteredPosts.length === 0` and you already check `filteredPosts.length > 0`, `filteredPosts.length >== 0` should cover both cases and you can always use fileredPosts to override the variable

Reply:
`filteredPosts.length >= 0` behaves differently when `posts.length > 0` and `filteredPosts.length === 0`, if I didn't overlook something. This distinction can be important if another unrelated post is deleted from the Communication section and it may cause our plagiarism post to be unintentionally removed from the `posts` array.

Reply:
You're right. Could you add a small comment above that statement, which explains it?

Reply:
I added it at https://github.com/ls1intum/Artemis/pull/5528/commits/2fa496fdd5f40be0aa22b93985dc6e1f4af4f4d3